scmrepo/progress.py

```diff
@@ -50,7 +50,7 @@ def __init__(self, fn) -> None:
 
     def __call__(self, msg: Union[str, bytes]) -> None:
         self._reporter._parse_progress_line(
-            msg.decode("ascii").strip() if isinstance(msg, bytes) else msg
+            msg.decode("utf-8").strip() if isinstance(msg, bytes) else msg
```

Comment:
hmm, probably we should use [`locale.getpreferredencoding()`](https://docs.python.org/3/library/locale.html#locale.getpreferredencoding) here? Let me look at dulwich.

Reply:
On my system, `locale.getpreferredencoding()` returns 'UTF-8`.
I'll change the code to use that.

Reply:
The message comes from the server, and depends on the server encoding: https://github.com/dulwich/dulwich/issues/687#issuecomment-462747144

For the purposes of this PR I think we should just assume that any typical server implementation is going to be returning `utf-8` (the local environment's `locale.getpreferredencoding()` is not what we want here) @kwon-young 

In the future we should consider how to handle not being able to decode the progress messages (and maybe just display some generic message) @skshetry

Reply:
@pmrowla, we could use `.decode(errors=error)` with `replace` or `surrogateescape`.

Reply:
> The message comes from the server, and depends on the server encoding: dulwich/dulwich#687 (comment)

Ho I did not know that.

I'll change back to utf-8 with `errors='replace'` (I suppose `surrogateescape` does not really matter since it is only a progress message)